sharedstore/getter.go

```diff
@@ -30,6 +35,10 @@ type promiseGetter struct {
 	promise lockmap.Promise
 }
 
+func (g *promiseGetter) WouldWait() bool {
+	return true
```

Comment:
Have I misunderstood something, or should this be
```suggestion
	select {
	case <-g.promise:
		return false
	case <-ctx.Done():
		return false
	default:
		return true
	}
```

Reply:
That is superior indeed, thanks :)